server/api/logout.js

```diff
@@ -17,11 +17,13 @@ exports.register = function (server, options, next) {
         },
         handler: function (request, reply) {
 
+            if (!request.auth.credentials || !request.auth.credentials.user) {
+                return reply({ message: 'Session not found.' }).takeover().code(404);
+            }
+
             var Session = request.server.plugins['hapi-mongo-models'].Session;
-            var credentials = request.auth.credentials || { user: {} };
-            var user = credentials.user || {};
             var query = {
-                username: user.username
+                userId: request.auth.credentials.user._id.toString()
             };
```

Comment:
Maybe it's best to follow suite from Aqua here?
https://github.com/jedireza/aqua/blob/master/server/api/logout.js#L25-L41